File: api/src/main/java/com/pedidosya/api/dto/Request/AddressDTO.java
package com.pedidosya.api.dto.Request;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.pedidosya.api.models.City;
import jakarta.persistence.Column;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.locationtech.jts.geom.Point;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class AddressDTO {
    private Integer idAddress;
    @NotNull
    @NotEmpty
    @Column(length =300 )
    private String description;
    //@NotNull
    //private Point coordinates;

    private Double latitude;

    private Double longitude;

    private Integer idCity;
}

File: dev-front/src/app/models/addressDto.ts
export interface AddressDTO {
    idClient: number,
    address: {
        idAddress: number,
        description: string,
        latitude: number,
        longitude: number,
        idCity: number
    },
    set: boolean
}

export interface AddressPostDTO {
    idClient: number,
    address: {
        description: string,
        latitude: number,
        longitude: number,
        idCity: number
    },
    set: boolean
}

export interface AddressGet {
    idClient: number
}

export interface Address {
    address: {
        idAddress: number,
        description: string,
        latitude: DoubleRange,
        longitude: DoubleRange,
        idCity: number
    },
}